plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -113,9 +116,19 @@ public BigQueryClient(
                 .expireAfterWrite(metadataCacheTtl.toMillis(), MILLISECONDS)
                 .shareNothingWhenDisabled()
                 .build(CacheLoader.from(this::listDatasetIdsFromBigQuery));
+        this.remoteDatasetCaseInsentiveCache = buildCache(caseInsensitiveNameMatchingCacheTtl);
```

Comment:
Do we need two separate caches if they have same ttl?

Reply:
Do you mean merging `remoteDatasetCaseInsentiveCache` & `remoteDatasetIdCache` into one?

Reply:
remoteDatasetCaseInsentiveCache & remoteTableCaseInsentiveCache